third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -122,6 +122,66 @@ Value shflIdxSync(Location loc, ConversionPatternRewriter &rewriter, Value val,
                         i32_val(0x1f));
 }
 
+Value storeShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
+                  Value val, Value pred) {
+  auto ty = val.getType();
+  auto val_ty = vec_ty(ty, 1);
+  Value vec_val = undef(val_ty);
+  vec_val = insert_element(val_ty, vec_val, val, i32_val(0));
+
+  auto vec_ty = vec_ty(i1_ty, 1);
+  Value vec_pred = undef(vec_ty);
+  vec_pred = insert_element(vec_ty, vec_pred, pred, i32_val(0));
+
+  rewriter.create<LLVM::MaskedStoreOp>(loc, vec_val, ptr, vec_pred, 4);
+  return val;
+}
+
+template<class T>
+T getVal(const std::string& val) {
```

Comment:
Also it's not clear to me why we'd return an enum here instead of just returning the mlir::Value that we want to be the default value?  That is, MLIR already has the ability to represent values, so can we just...use that here instead of making a new way of representing values?